src/main/java/duke/ChatBot.java

```diff
@@ -0,0 +1,69 @@
+package duke;
+
+import java.io.IOException;
+
+import duke.command.Command;
+import duke.exception.NoDataFileException;
+
+/**
+ * The chat bot class to handle the internal logic.
+ */
+public class ChatBot {
+    private final Ui ui;
+    private final Parser parser;
+    private TaskList tasks;
+
+    /**
+     * Constructor of the chat bot.
+     * Construct a new chat bot with new UI, and initialised storage handling.
+     * Try to load the previous data file, if failed create a new data file.
+     * If Duke cannot create a new one, <code>IOException</code> will be caught
+     * and prints information for the user.
+     */
+    public ChatBot() {
+        ui = new Ui();
+        ui.showLoading();
+        parser = new Parser();
+        try {
+            tasks = new TaskList(Storage.readList());
+            ui.showLoaded(tasks.size());
+        } catch (NoDataFileException e) {
+            tasks = new TaskList();
+            ui.showLoadedNew();
+        } catch (IOException e) {
+            ui.showLoadingError();
+        }
+    }
+
+    /**
+     * The entry point to run the chat bot.
+     */
+    public void start() {
+        ui.welcome();
+        boolean isExit = false;
+        while (!isExit) {
+            try {
+                String response = ui.receiveChat();
+                ui.printHorizontal();
+                Command cmd = parser.parse(response);
+                cmd.execute(tasks, ui);
+                Storage.save(tasks.getList());
+                isExit = cmd.isExit();
+            } catch (Exception e) {
+                ui.printLine(e.getMessage());
+            }
+            ui.printHorizontal();
+        }
+    }
+
+    public String getResponse(String input) {
+        try {
+            Command cmd = parser.parse(input);
+            cmd.execute(tasks, ui);
+            Storage.save(tasks.getList());
+            return cmd.getMessage();
+        } catch (Exception e) {
+            return e.getMessage();
```

Comment:
I agree with the above comment that ideally you should name out the exceptions and handle each exception seperately.